src/or/voting_schedule.c

```diff
@@ -138,14 +142,32 @@ voting_schedule_t voting_schedule;
 time_t
 voting_schedule_get_next_valid_after_time(void)
 {
+  bool need_to_recalculate_voting_schedule = false;
+
   /* This is a safe guard in order to make sure that the voting schedule
    * static object is at least initialized. Using this function with a zeroed
    * voting schedule can lead to bugs. */
   if (tor_mem_is_zero((const char *) &voting_schedule,
                       sizeof(voting_schedule))) {
-    voting_schedule_recalculate_timing(get_options(), time(NULL));
+    need_to_recalculate_voting_schedule = true;
+  }
+
+  /* Also make sure we are not using an outdated voting schedule. If we have a
+   * newer consensus, make sure we recalculate the voting schedule. */
+  networkstatus_t *consensus = networkstatus_get_live_consensus(approx_time());
+  if (consensus &&
+      consensus->valid_after != voting_schedule.live_consensus_valid_after) {
```

Comment:
Maybe we want to use a `else if` somehow so we avoid all this code if the first condition was hit which would recalculate the voting schedule in the first place?

Reply:
Good point. Took the goto approach in 82bb4f34 to avoid the extra indentation of `else if`. Let me know if you hate it, and I will just use `else if`.